BCStudentSoftwareDevTeam/celts
# Selectively remove service transcript entry for banned programs

Issue: #658 
Add a checkbox to the ban modal dialog to choose whether to completely remove the program from the service transcript or keep it in the service transcript (in case if the checkbox is not clicked). The checkbox on the modal should affect the service transcript directly and remove or keep the programs, whose modals were checked/unchecked.

Did:

- Moved the checkbox outside the modal on its own column "On Trancript"
- Added a new boolean column `removeFromTranscript` in programBan table 
- Added a new route to update programBan table when checkbox state changes
- The fixes are reflected on the database and the service transcript page
- Added a message next to the checkbox letting users know if they "added" or "removed" the program from transcript whenever they check/uncheck it
- Modified the getTotalHours function in transcript.py to calculate hours correctly for banned/unbanned programs
- Made the unbanned programs show on the transcript by default (checked and greyed out)

Test:

- checkout the branch `serviceTEntry658`
- reset the database and run flask
- Go to student search and search for a student, like "khatts"
- In the Choose Action dropdown, select view service transcript, note which programs appear on the transcript
- Go back to the student profile, and under "Programs" in accordion, click on the pencil edit button that will open a modal Ban/Unban
- Ban a student from another program like "Hunger Initiatives" (keep in mind that the program will show up in the service transcript only if the student earned hours for that program)
- Now that the program is banned, you can now modify the checkmark. Try to uncheck the On Transcript box and see if the message "removed" appears
- Go back to view the service transcript, and check if that program appears (it should not appear). Also make sure that the Total Event Hours is correct
- Try to check the checkbox and see if the messaged "added" appears and if the program appears again on the service transcript. Also check the hours again to make sure it is correct
- Review Code in these files: userProfile.js, userProfile.html, main/routes.py, transcript.py
